File: symbol.go
package main

import (
	"github.com/paidgeek/ssdl/token"
)

type Position struct {
	Line   int
	Column int
}

func NewPosition(line int, column int) Position {
	return Position{
		Line:line,
		Column:column,
	}
}

type Symbol struct {
	Token    token.Token
	Lexeme   string
	Position Position
}

func NewSymbol(token token.Token, lexeme string, line int, column int) Symbol {
	return Symbol{
		Token:token,
		Lexeme:lexeme,
		Position:NewPosition(line, column),
	}
}

File: src/main/java/com/moybl/ssdl/Symbol.java
package com.moybl.ssdl;

public class Symbol {

	private Token token;
	private String lexeme;
	private Position position;

	public Symbol(Token token, String lexeme, Position position) {
		this.token = token;
		this.lexeme = lexeme;
		this.position = position;
	}

	public Symbol(Token token, String lexeme, int startLine, int endLine, int startColumn, int endColumn) {
		this.token = token;
		this.lexeme = lexeme;
		this.position = new Position(startLine, endLine, startColumn, endColumn);
	}

	public Symbol(Token token, String lexeme, int line, int startColumn, int endColumn) {
		this.token = token;
		this.lexeme = lexeme;
		this.position = new Position(line, line, startColumn, endColumn);
	}

	public Token getToken() {
		return token;
	}

	public void setToken(Token token) {
		this.token = token;
	}

	public String getLexeme() {
		return lexeme;
	}

	public void setLexeme(String lexeme) {
		this.lexeme = lexeme;
	}

	public Position getPosition() {
		return position;
	}

	public void setPosition(Position position) {
		this.position = position;
	}

}
